homeassistant/components/tplink/entity.py

```diff
@@ -153,12 +146,11 @@ def __init__(
             }
 
         self._attr_unique_id = self._get_unique_id()
-        self._attr_entity_category = self._category_for_feature(feature)
 
     def _get_unique_id(self) -> str:
         """Return unique ID for the entity."""
         device = self._device
-        if self._feature is not None:
+        if not isinstance(self, SensorEntity) and self._feature is not None:
```

Comment:
the base entity should never know about the details of a platform

Reply:
One way around this is to overload `_get_unique_id` in the subclass based on the platform

Reply:
Done in https://github.com/home-assistant/core/pull/120025